store/src/java-test/com/zimbra/cs/mailbox/DumpsterTest.java

```diff
@@ -144,9 +142,7 @@ public void moveDumpsterEmail() throws Exception {
             if (!immutableException) {
                 throw e;
             }
-        } finally {
-            mbox.endTransaction(success);
```

Comment:
You removed a call to `endTransaction`? But there is no corresponding transaction commit/close in the new code? Please explain why this is the correct change to make here.

Reply:
Added assertion to check to see if there is a transaction that needs to be cleaned up here.